### PR TITLE
Add resource and DB model for contracts/filters

### PR DESCRIPTION
**Pull request by amittbose**
Signed-off-by: Amit Bose <amitbose@gmail.com>

### CODE REVIEW

File: aim/api/resource.py [hunk 1]
```diff
@@ -256,3 +256,113 @@ def __init__(self, **kwargs):
...
+
+
+class FilterEntry(AciResourceBase):
+    """Resource representing a classifier entry of a filter in ACI.
+
+    Identity attributes: name of ACI tenant, name of filter and name of entry.
+
+    Values for classification fields may be integers as per standards
+    (e.g. ip_protocol = 6 for TCP, 17 for UDP), or special strings listed
+    below. UNSPECIFIED may be used to indicate that a particular
```
Comment (ivar-lazzaro) [inline aim/api/resource.py]: Is a fully "unspecified" filter working as a "allow all"?

Comment (amittbose) [inline aim/api/resource.py]: That's right - an entry like FilterEntry(tenant_name=..., filter_name=..., name=...) is an allow-all entry.

File: aim/db/models.py [hunk 2]
```diff
@@ -162,3 +162,120 @@ def to_attr(self, session):
...
+
+
+class Filter(model_base.Base, model_base.HasAimId,
+             model_base.HasName, model_base.HasDisplayName,
+             model_base.HasTenantName, model_base.AttributeMixin):
+    """DB model for Filter."""
+
+    __tablename__ = 'aim_filters'
+    __table_args__ = (uniq_column(__tablename__, 'tenant_name', 'name') +
+                      to_tuple(model_base.Base.__table_args__))
```
Comment (ivar-lazzaro) [inline aim/db/models.py]: would a Entry backref be useful here?

Comment (amittbose) [inline aim/db/models.py]: I don't see the need yet. This is a simple parent-child relationship in the tree, and so far there hasn't been a need to create a backref to the parent from the child.

File: aim/api/resource.py (hunk 1 above)
Comment (ivar-lazzaro) [inline aim/api/resource.py]: sounds good

File: aim/db/models.py (hunk 2 above)
Comment (ivar-lazzaro) [inline aim/db/models.py]: yeah I guess only faults and status have such a thing, no need to add a backref if we don't believe there's a use for that.

**Pull request merged**